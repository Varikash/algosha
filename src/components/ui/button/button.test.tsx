import React from "react";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe('Button component', () => {
  it('Отрисовывает кнопку с текстом', () => {
    const component = renderer.create(<Button text="Развернуть" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Отрисовывает кнопку без текста', () => {
    const component = renderer.create(<Button  text=""/>)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Отрисовывает неактивную кнопку', () => {
    const component = renderer.create(<Button text="Развернуть" disabled={true} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Отрисовывает кнопку с loader', () => {
    const component = renderer.create(<Button text="Развернуть" isLoader={true} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Триггерит колбэк при нажатии на кнопку', () => {
    window.alert = jest.fn();
    render(<Button text="Развернуть" onClick={() => alert('Кнопка нажата')} />);

    fireEvent.click(screen.getByText('Развернуть'));
    expect(window.alert).toHaveBeenCalledTimes(1);
  });




})