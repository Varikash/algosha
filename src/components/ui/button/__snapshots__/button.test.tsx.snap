// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Отрисовывает кнопку без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component Отрисовывает кнопку с loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component Отрисовывает кнопку с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;

exports[`Button component Отрисовывает неактивную кнопку 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;
